src/main.rs

```diff
@@ -489,9 +468,26 @@ mod test_song_planning {
 }
 
 fn main() {
-	let args: Vec<String> = env::args().collect();
+	// let args: Vec<String> = env::args().collect();
+
+	// Get executable file location or default to `./` if failed.
+	let mut songs_directory = env::current_exe()
+		.unwrap_or(PathBuf::from("./stream_autodj"));
+
+	// env::current_exe() might return path to symbolic link on some platforms.
+	// try to get real path if it's a symbolic link
+	songs_directory = match songs_directory.read_link() {
+		Ok(path) => path,
+		Err(_) => songs_directory,
+	};
+
+	// Remove executable file name from the path.
+	songs_directory.pop();
+
+	// add 'songs' to the end of the path.
+	songs_directory.push("songs");
 
-	let paths = fs::read_dir("./songs").unwrap();
+	let paths = fs::read_dir(songs_directory).unwrap();
```

Comment:
Please change this back to the way it was, it's intentional.

Reply:
Yeah, that's what I'm doing now.